galaxy_ng/app/api/v3/viewsets/collection.py

```diff
@@ -180,3 +186,64 @@ def get(self, request, *args, **kwargs):
         metrics.collection_artifact_download_failures.labels(status=response.status_code).inc()
         raise APIException('Unexpected response from content app. '
                            f'Code: {response.status_code}.')
+
+
+class CollectionVersionMoveViewSet(ViewSet):
+    permission_classes = GALAXY_PERMISSION_CLASSES + [
+        permissions.IsNamespaceOwner
+    ]
+
+    def move_content(self, request, *args, **kwargs):
+        """Remove content from source repo and add to destination repo.
+
+        Creates new RepositoryVersion of source repo without content included.
+        Creates new RepositoryVersion of destination repo with content included.
+        """
+
+        try:
+            collection_version = CollectionVersion.objects.get(
+                namespace=self.kwargs['namespace'],
+                name=self.kwargs['name'],
+                version=self.kwargs['version'],
+            )
+        except ObjectDoesNotExist:
+            raise NotFound('Collection version not found')
+
+        try:
+            src_repo = AnsibleDistribution.objects.get(
+                base_path=self.kwargs['source_path']).repository
+            dest_repo = AnsibleDistribution.objects.get(
+                base_path=self.kwargs['dest_path']).repository
+        except ObjectDoesNotExist:
+            raise NotFound('Repo(s) for moving collection not found')
+
+        src_versions = CollectionVersion.objects.filter(pk__in=src_repo.latest_version().content)
+        if collection_version not in src_versions:
+            raise NotFound('Collection version not found in source repo')
+
+        dest_versions = CollectionVersion.objects.filter(pk__in=dest_repo.latest_version().content)
+        if collection_version in dest_versions:
+            raise APIException('Collection version already found in destination repo')
+
+        add_task = self._add_content(collection_version, dest_repo)
+        remove_task = self._remove_content(collection_version, src_repo)
+
+        return Response(
+            data={
+                'add_task_id': add_task.id,
+                'remove_task_id': remove_task.id,
```

Comment:
Where would these task_id's be looked up?  Is there the equilivent of the the /imports/ endpoints or would client tooling need to use the pulp api task endpoints to check status?

Reply:
i think it would need to use a pulp api task endpoint. client should not need to check status for the same reasons as an import which could take a long time - this operation should be quick unless there is a system issue like locked repo or all workers busy.

Reply:
I don't know of a particular use case for the task_ids currently. I suspect the first thing that would want/need the task_ids and task status would be web ui stuff just as progress indicators.

Mainly just want to verify that data is returned in a useful format which for this case would just mean the task_ids can be used to find the task status if needed. Particularly if the tasks fail for some reason.

Reply:
yeah these are the same task id formats as our other pulp tasks